Implement config file structure for device/vlan sharing

Gandem opened the pull request:
Implements a config file structure with two views : 

- By pool, to know to which pool we should forward requests.
- By Bonjour server mac address, to know to which pool we should forward responses.

 @KSerrania @oxlay Any take on that, if someone feels like to review that code ? (I should probably add tests... :see_no_evil:)

oxlay commented:
Lgtm 👍 added tests in [another PR](https://github.com/Gandem/bonjour-reflector/pull/4) (to merge in `nayef/config-views`)

KSerrania commented on config.toml:
```diff
@@ -1 +1,30 @@
+description = "Test Chromecast"
+mac_address = "AA:BB:CC:DD:EE:FF"
+pools = [1078, 3597]
+pools_description = [
```
Idea: add a separate `[pools]` table where we put all the details about a pool (owner, description, ...). This way we don't repeat the same things about the pools for each device, and we also don't have to check that the `pools` and `pools_description` arrays are of the same size (we do have to check for the presence of a pool in the resulting map, however).
```toml
[[devices]]
description = "Test Chromecast"
mac_address = "AA:BB:CC:DD:EE:FF"
pools = [1078, 3597]

[pools]
    [pools.1078]
    description = "John Doe"

    [pools.1547]
    description = "Franck Dupond"

    [pools.2483]
    description = "Jean Poullain"

    [pools.3133]
    description = "Jean-Jacques Yves du Périgorac"

    [pools.3597]
    description = "Jane Doe"
```

```go
type brconfig struct {
	NetInterface string          `toml:"net_interface"`
	Devices      []bonjourDevice `toml:"devices"`
	Pools        map[string]pool `toml:"pools"`
}

type pool struct {
	Description string `toml:"description"`
}
```
Note: with this, the `Pools` map keys are strings, it would be better if they were integers

oxlay replied:
@Gandem is the `description` field intended to be used somewhere in the code or is it only there to improve the readability of the config file?

Gandem replied:
Only to improve readability

KSerrania commented:
Lgtm too :+1:

Gandem commented:
I rewrote the config file structure and split the `origin_pool` from all the other `shared_pools`. The naming convention is as follows:

- The *origin pool* is supposed to be the pool in which the Bonjour **server** (chromecast/spotify air/etc...) lives. 

- The *shared pools* are the pools in which the **clients** live.

This way, Bonjour **requests** are not forwarded to all the pools without distinction (as first planned), but only to the origin pool. By doing this, we also make the assumption that the bonjour devices will not move frequently between vlan pools. I rewrote `mapByPool` accordingly.

Regarding the pools description field, I was planning to have it mainly for readability purposes, (to make sure sharing config don't stay here eternally, etc...) but following your comments I realized that this is easier to achieve with comments. Maybe we should add an expiry date for each field, or even better interface this with Webaz, but this is clearly not a priority :see_no_evil:

oxlay commented:
The new approach (declaring pools separately) seems better to me  👍 
I also think we should leave pools description as (encouraged) comments

Gandem commented:
Refactor config to a map, as suggested by @KSerrania so the config would already be mapped by address (mac addresses are supposed to be unique)
